modules/apps/foundation/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/portlet/action/UpdateRolesMVCActionCommand.java

```diff
@@ -0,0 +1,154 @@
+@Component(
+	immediate = true,
+	property = {
+		"javax.portlet.name=" + UsersAdminPortletKeys.MY_ACCOUNT,
+		"javax.portlet.name=" + UsersAdminPortletKeys.MY_ORGANIZATIONS,
+		"javax.portlet.name=" + UsersAdminPortletKeys.USERS_ADMIN,
+		"mvc.command.name=/users_admin/update_roles"
+	},
+	service = MVCActionCommand.class
+)
+public class UpdateRolesMVCActionCommand extends BaseMVCActionCommand {
+
+	@Override
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		try {
+			User user = _portal.getSelectedUser(actionRequest);
+
+			Contact contact = user.getContact();
+
+			Calendar birthdayCal = CalendarFactoryUtil.getCalendar();
+
+			birthdayCal.setTime(user.getBirthday());
+
+			long[] roleIds = _usersAdmin.getRoleIds(actionRequest);
+
+			List<UserGroupRole> userGroupRoles = null;
+
+			String addGroupRolesGroupIds = actionRequest.getParameter(
+				"addGroupRolesGroupIds");
+			String addGroupRolesRoleIds = actionRequest.getParameter(
+				"addGroupRolesRoleIds");
+			String deleteGroupRolesGroupIds = actionRequest.getParameter(
+				"deleteGroupRolesGroupIds");
+			String deleteGroupRolesRoleIds = actionRequest.getParameter(
+				"deleteGroupRolesRoleIds");
+
+			if ((addGroupRolesGroupIds != null) ||
+				(addGroupRolesRoleIds != null) ||
+				(deleteGroupRolesGroupIds != null) ||
+				(deleteGroupRolesRoleIds != null)) {
+
+				userGroupRoles = _usersAdmin.getUserGroupRoles(actionRequest);
+			}
+
+			ServiceContext serviceContext = ServiceContextFactory.getInstance(
+				User.class.getName(), actionRequest);
+
+			_userService.updateUser(
+				user.getUserId(), user.getPassword(), null, null,
+				user.getPasswordReset(), null, null, user.getScreenName(),
+				user.getEmailAddress(), user.getFacebookId(), user.getOpenId(),
+				user.getLanguageId(), user.getTimeZoneId(), user.getGreeting(),
+				user.getComments(), user.getFirstName(), user.getMiddleName(),
+				user.getLastName(), contact.getPrefixId(),
+				contact.getSuffixId(), user.isMale(),
+				birthdayCal.get(Calendar.MONTH), birthdayCal.get(Calendar.DATE),
+				birthdayCal.get(Calendar.YEAR), contact.getSmsSn(),
+				contact.getFacebookSn(), contact.getJabberSn(),
+				contact.getSkypeSn(), contact.getTwitterSn(),
+				user.getJobTitle(), user.getGroupIds(),
+				user.getOrganizationIds(), roleIds, userGroupRoles,
+				user.getUserGroupIds(), serviceContext);
+		}
+		catch (Exception e) {
+			if (e instanceof NoSuchUserException ||
+				e instanceof PrincipalException) {
+
+				SessionErrors.add(actionRequest, e.getClass());
+
+				actionResponse.setRenderParameter("mvcPath", "/error.jsp");
+			}
+			else if (e instanceof MembershipPolicyException) {
+				SessionErrors.add(actionRequest, e.getClass(), e);
+
+				actionRequest.setAttribute(
+					UsersAdminWebKeys.ACTION_COMMAND_NAME,
+					"/users_admin/update_roles");
+				actionRequest.setAttribute(
+					UsersAdminWebKeys.JSP_PATH, "/user/roles.jsp");
```

Comment:
Is it necessary to set these two attributes if we're already setting the screen navigation parameters below?  Won't their classes take care of it?

Reply:
That's right. We don't need these two attributes set. I'll make the change and update the branch. Thanks for catching it!

Reply:
👍